optuna/integration/allennlp.py

```diff
@@ -373,6 +373,14 @@ def run(self) -> float:
         for package_name in self._include_package:
             allennlp.common.util.import_module_and_submodules(package_name)
 
+        # Without the following lines, the transformer model construction only takes place in the
+        # first trial (which would consume some random numbers), and the cached model will be used
+        # in trials afterwards (which would not consume random numbers), leading to inconsistent
+        # results between single trial and multiple trials. To make results reproducible in
+        # multiple trials, we clear the cache before each trial.
+        allennlp.common.cached_transformers._model_cache.clear()
```

Comment:
[note]
`allennlp.common.cached_transformers._clear_caches` would be introduced in the next release of AllenNLP.

https://github.com/allenai/allennlp/commit/a463e0e74f4908950ccdfe35198d13f38de6ae1b#diff-52c88d0594aac08ef176c393bdd7cb32ba8f57c6cc7f9a39583e741100fa4309R146-R153

Reply:
Yes, but it's currently not available. Should we use the new API and update dependency to the next AllenNLP release?

Reply:
No, it's reasonable to keep invoking `xxx_cache.clear()` directly. When dropping a current version of AllenNLP, I'll revise the implementation. :+1:

Reply:
Please make a comment of when this can be removed for future maintainers. 🙇

Reply:
Ok, I add a TODO for this